### PR TITLE
_gridded_eos_mm.py reader for gridded NASA EOS hdf files

### PR DESCRIPTION
**Pull request by dwfncar**
Initial reader for gridded NASA EOS datasets, which include MODIS level 3 products.
Uses monetio.sat.hdfio.
Similar to monetio/sat/_modis_l2_mm.py.

### CODE REVIEW

**Comment by dwfncar** [comment]
Will fix pre-commit lint checks, mostly single quotes need to be replaced by double quotes

**Comment by zmoon** [comment]
@dwfncar is this ready to go?

**Comment by dwfncar** [comment]
Yes, ready for review and merge if OK ...
David

File: monetio/sat/_gridded_eos_mm.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import logging
+
+import numpy as np
+import xarray as xr
+
+
+def read_gridded_eos(fname, var_dict, save_as_netcdf=False):
```
Comment (zmoon) [inline monetio/sat/_gridded_eos_mm.py]:
```suggestion
def read_gridded_eos(fname, var_dict, *, save_as_netcdf=False):
```

File: monetio/sat/_gridded_eos_mm.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    Parameters
+    __________
+    fname : str
+        Input file path.
+    var_dict : dict
+        Dictionary of variables to read, along with variable metadata.
+        {varname: {"fillvalue": float, "scale": float, "units": str}, ...}
+    save_as_netcdf: bool, default=False
```
Comment (zmoon) [inline monetio/sat/_gridded_eos_mm.py]:
```suggestion
    save_as_netcdf : bool, default=False
```

File: monetio/sat/_gridded_eos_mm.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        Save variables in var_dict in netcdf format.
+
+
+    Returns
+    _______
+    xarray.Dataset
+    """
+
+    from monetio.sat.hdfio import hdf_close, hdf_list, hdf_open, hdf_read
```
Comment (zmoon) [inline monetio/sat/_gridded_eos_mm.py]:
```suggestion
    from .hdfio import hdf_close, hdf_list, hdf_open, hdf_read
```

File: monetio/sat/_gridded_eos_mm.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    ds_dict = dict()
+
+    logging.info("read_gridded_eos:" + fname)
+    f = hdf_open(fname)
+    datasets, indices = hdf_list(f)
+
+    lon = hdf_read(f, "XDim")
+    lat = np.flip(hdf_read(f, "YDim"))
+    lon_da = xr.DataArray(lon, attrs={"longname": "longitude", "units": "deg East"})
+    lat_da = xr.DataArray(lat, attrs={"longname": "latitude", "units": "deg North"})
```
Comment (zmoon) [inline monetio/sat/_gridded_eos_mm.py]:
```suggestion
    lon_da = xr.DataArray(lon, attrs={"long_name": "longitude", "units": "deg east"})
    lat_da = xr.DataArray(lat, attrs={"long_name": "latitude", "units": "deg north"})
```

File: monetio/sat/_gridded_eos_mm.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+def read_gridded_eos(fname, var_dict, save_as_netcdf=False):
+    """
+    Parameters
+    __________
+    fname : str
+        Input file path.
+    var_dict : dict
+        Dictionary of variables to read, along with variable metadata.
+        {varname: {"fillvalue": float, "scale": float, "units": str}, ...}
```
Comment (zmoon) [inline monetio/sat/_gridded_eos_mm.py]:
```suggestion
        ``{varname: {"fillvalue": float, "scale": float, "units": str}, ...}``
```

**Review by zmoon — COMMENTED** [review]
Looks fine to me. I would also suggest creating a logger with `logger = logging.getLogger(__name__)` and using that for the logging, that way it shows where the message is coming from.

Also if you want to add it to `sat/__init__.py` you could, I think we did that last time.

**Comment by zmoon** [comment]
For next time, if you want, you [can use GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) to commit selected suggestions as a commit automatically. And I think they auto mark as resolved then too.

**Comment by zmoon** [comment]
@dwfncar if you want to confirm that this still works for you, I'll merge this.

**Comment by dwfncar** [comment]
Yes, tested reader with MODIS L3 data.  We can proceed with the merge.

**Pull request merged**